### PR TITLE
Refactor Symbolic Expr

### PR DESCRIPTION
**Pull request by tobiasduerschmid**

### CODE REVIEW

File: include/rosdiscover-clang/Ast/Stmt/ApiCall.h [hunk 1]
```diff
@@ -117,31 +122,29 @@ class RateSleep : public SymbolicRosApiCall {
 
 class Publish : public SymbolicRosApiCall {
 public:
-  Publish(std::string const &publisher, std::vector<std::unique_ptr<SymbolicControlDependency>> controlDependencies={}) : 
+  Publish(std::string const &publisher, std::unique_ptr<SymbolicExpr> controlDependencies = std::make_unique<BoolLiteral>(true)) :
```
Comment (ChrisTimperley) [inline include/rosdiscover-clang/Ast/Stmt/ApiCall.h:125]: What does it mean for control dependencies to be the literal "true"? Control dependencies suggests a collection of some item, whereas the SymbolicExpr suggests something else. Renaming this argument (for all of the relevant classes) might be helpful.

File: include/rosdiscover-clang/Value/String.h [hunk 2]
```diff
@@ -16,13 +16,17 @@ class StringLiteral : public virtual SymbolicString {
   }
 
   void print(llvm::raw_ostream &os) const override {
-    os << "\"" << literal << "\"";
+    os << toString();
+  }
+  std::string toString() const override {
+    return "'\"" + literal + "\"'";
```
Comment (ChrisTimperley) [inline include/rosdiscover-clang/Value/String.h:22]: this is fine, but for the future, look into using `fmt::format` (https://github.com/fmtlib/fmt)

Comment (tobiasduerschmid) [reply include/rosdiscover-clang/Value/String.h]: fixed

File: include/rosdiscover-clang/Ast/Stmt/ApiCall.h (hunk 1 above)
Comment (tobiasduerschmid) [reply include/rosdiscover-clang/Ast/Stmt/ApiCall.h]: fixed

**Pull request merged**